Update to latest patch release of Ceph for CVE fixes

MikaelSmith opened the pull request.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/replicatedhq/kURL?pullRequest=1653) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/replicatedhq/kURL?pullRequest=1653) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/replicatedhq/kURL?pullRequest=1653) it.</sub>

MikaelSmith commented:
I've tested this out locally using the ceph image directly and rebuilding the rook-ceph image. I updated the ceph image in the CephCluster for an existing 1.0.4 install, and updated the rook-ceph-operator deployment with the new rook-ceph image.

Rebuilding rook-ceph was necessary to update the client so it could talk to the server after also running `ceph config set mon auth_allow_insecure_global_id_reclaim false`.

I'd need help with the rest of the steps in https://github.com/replicatedhq/kURL/blob/master/addons/rook/1.0.4/build-images/README.md.

MikaelSmith commented:
It's not yet clear to me whether new installs need to set `mon_warn_on_insecure_global_id_reclaim_allowed = false`.

MikaelSmith commented on addons/rook/1.0.4/build-images/ceph/Dockerfile:
```diff
@@ -1,6 +1,6 @@
 # https://github.com/ceph/ceph-container
 
-FROM ceph/ceph:v14.2.0
+FROM ceph/ceph:v14.2.21
```
It probably makes sense to add this as a new addon version like `1.0.4-14.2.21`. Image tag would also need to change.

emosbaugh commented:
Resolved by  #1726 and #1742
